workspaces/adventure-pack/goodies/typescript/Iterator.prototype.drop/index.test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).drop;
+
+import "./index";
+
+describe("Iterator.prototype.drop", () => {
+  it("drops the specified number of elements from the iterator", () => {
+    const result = Array.from([10, 22, 33, 47, 58].values().drop(3));
+    expect(result).toStrictEqual([47, 58]);
+  });
+
+  it("returns an empty iterator if the specified number is greater than the iterator length", () => {
+    const result = Array.from([10, 22, 33].values().drop(5));
```

Comment:
We don't have to go crazy with this, but it would be nice to have slightly more variety in the kinds of iterators we use. For example, an iterator created from a `Set`, or the characters of a string (which can also be iterated over). See the tests for `Iterator.prototype.map` for a bunch of things we could do.

Also, for the cases where we use numbers in the array, let's try to not always have them be increasing. This is the kind of pattern that is easily avoided to make the test slightly more robust. 

In my opinion, test writing is a bit like security. Imagine someone's goal was to write an incorrect implementation of the code that passed all the tests. How can I make that more difficult for them? Avoiding obvious patterns in the test data is low enough effort to be worth adopting.

Reply:
agreed!

Reply:
Going off from @miorel yeah it'd be cool to see an iterator from `Map` entries